x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/spi/SpatialExtension.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.spatial.spi;
+
+import org.elasticsearch.Version;
+import org.elasticsearch.common.xcontent.support.XContentMapValues;
+import org.elasticsearch.geometry.Geometry;
+import org.elasticsearch.index.mapper.AbstractGeometryFieldMapper;
+import org.elasticsearch.index.mapper.GeoShapeFieldMapper;
+import org.elasticsearch.index.mapper.GeoShapeIndexer;
+import org.elasticsearch.index.mapper.MappedFieldType;
+import org.elasticsearch.index.mapper.ParseContext;
+import org.elasticsearch.index.mapper.TypeParsers;
+import org.elasticsearch.index.query.VectorGeoShapeQueryProcessor;
+import org.elasticsearch.license.XPackLicenseState;
+import org.elasticsearch.xpack.core.XPackPlugin;
+
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+public class SpatialExtension implements GeoShapeFieldMapper.Extension, AbstractGeometryFieldMapper.ParserExtension {
+
+    public SpatialExtension() {
+    }
+
+    public static XPackLicenseState getLicenseState() { return XPackPlugin.getSharedLicenseState(); }
+
+    @Override
+    public Map<String, GeoShapeFieldMapper.DataHandlerFactory> getDataHandlerFactories() {
+        XPackLicenseState licenseState = getLicenseState();
+        if (licenseState != null && licenseState.isSpatialAllowed()) {
+            // HACK: override the default data handler factory, this is trappy because other plugins could override
+            GeoShapeFieldMapper.DATA_HANDLER_FACTORIES.replace(
+                GeoShapeFieldMapper.Defaults.DATA_HANDLER.value(), () -> new SpatialDataHandler());
```

Comment:
what about this alternative approach:
 - have a default data handler in `server/` but don't register it, it is only used as a fallback if no plugins register a data handler
 - fail the node if there is more than one registered data handler
 - default doc values to true if, and only if there is one registered handler

Reply:
:+1:  I like that approach better since it doesn't involve squashing the data handler.